airflow/providers/cloudera/operators/cde_operator.py

```diff
@@ -0,0 +1,252 @@
+    submitted to the API or not.
+
+    Note: all parameters below can also be provided through the
+    ``default_args`` field of the DAG.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:CdeRunJobOperator`
+
+    :param job_name: the name of the job in the target cluster, required
+    :param connection_id: the Airflow connection id for the target API
+        endpoint, default value ``'cde_runtime_api'``
+    :param variables: a dictionary of key-value pairs to populate in the
+        job configuration, default empty dict.
+    :param overrides: a dictionary of key-value pairs to override in the
+        job configuration, default empty dict.
+    :param wait: if set to true, the operator will wait for the job to
+        complete in the target cluster. The task exit status will reflect the
+        status of the completed job. Default ``True``
+    :param timeout: the maximum time to wait in seconds for the job to
+        complete if ``wait=True``. If set to ``None``, 0 or a negative number,
+        the task will never be timed out. Default ``0``.
+    :param job_poll_interval: the interval in seconds at which the target API
+        is polled for the job status. Default ``10``.
+    :param api_retries: the number of times to retry an API request in the event
+        of a connection failure or non-fatal API error. Default ``9``.
+    """
+
+    template_fields = ("variables", "overrides")
+    ui_color = "#385f70"
+    ui_fgcolor = "#fff"
+
+    DEFAULT_WAIT = True
+    DEFAULT_POLL_INTERVAL = 10
+    DEFAULT_TIMEOUT = 0
+    DEFAULT_RETRIES = 9
+    DEFAULT_CONNECTION_ID = "cde_runtime_api"
+    DEFAULT_API_TIMEOUT = 30
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        job_name: str,
+        variables: dict[str, Any] | None = None,
+        overrides: dict[str, Any] | None = None,
+        connection_id: str = DEFAULT_CONNECTION_ID,
+        wait: bool = DEFAULT_WAIT,
+        timeout: int = DEFAULT_TIMEOUT,
+        job_poll_interval: int = DEFAULT_POLL_INTERVAL,
+        api_retries: int = DEFAULT_RETRIES,
+        api_timeout: int = DEFAULT_API_TIMEOUT,
+        user=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.job_name = job_name
+        self.variables = variables or {}
+        self.overrides = overrides or {}
+        self.connection_id = connection_id
+        self.wait = wait
+        self.timeout = timeout
+        self.job_poll_interval = job_poll_interval
+        if user:
+            self.log.warning("Proxy user is not yet supported. Setting it to None.")
+        self.user = None
+        self.api_retries = api_retries
+        self.api_timeout = api_timeout
+        if not self.job_name:
+            raise ValueError("job_name required")
+        # Set internal state
+        self._hook = self.get_hook()
```

Comment:
Seems like this might add additional Metadabase / Secrets Backend usage.
`CdeHook` itself obtain credentials during `__init__` so that mean operator also will get connection during each DagProcessor parsing loop.

Better to move this to cached property which only accessible from `execute` method, for example

https://github.com/apache/airflow/blob/430e930902792fc37cdd2c517783f7dd544fbebf/airflow/providers/amazon/aws/operators/athena.py#L98-L103

Reply:
Thanks for advice, will be investigated.